### PR TITLE
gh-117765: Improve documentation for `mocker.patch.dict`

### PR DESCRIPTION
**Pull request by GovernmentPlates**
* Issue: gh-117765

----
📚 Documentation preview 📚: https://cpython-previews--121755.org.readthedocs.build/

### CODE REVIEW

File: Doc/library/unittest.mock.rst [hunk 1]
```diff
@@ -1624,7 +1624,8 @@ patch.dict
 .. function:: patch.dict(in_dict, values=(), clear=False, **kwargs)
 
     Patch a dictionary, or dictionary like object, and restore the dictionary
-    to its original state after the test.
+    to its original state after the test, where the restored dictionary is the
```
Comment (ZeroIntensity) [inline Doc/library/unittest.mock.rst]:
```suggestion
    to its original state after the test, where the restored dictionary is a
```

File: Lib/unittest/mock.py [hunk 2]
```diff
@@ -1829,7 +1829,8 @@ def patch(
 class _patch_dict(object):
     """
     Patch a dictionary, or dictionary like object, and restore the dictionary
-    to its original state after the test.
+    to its original state after the test, where the restored dictionary is
+    the copy of the dictionary as it was before the test.
```
Comment (ZeroIntensity) [inline Lib/unittest/mock.py]:
```suggestion
    a copy of the dictionary as it was before the test.
```

File: Misc/NEWS.d/next/Documentation/2024-07-14-12-25-53.gh-issue-117765.YFMOUv.rst [hunk 3]
```diff
@@ -0,0 +1 @@
+Improve documentation for :func:`unittest.mock.patch.dict`
```
Comment (ZeroIntensity) [inline Misc/NEWS.d/next/Documentation/2024-07-14-12-25-53.gh-issue-117765.YFMOUv.rst]:
```suggestion
Improved documentation for :func:`unittest.mock.patch.dict`
```

**Review by ZeroIntensity — CHANGES_REQUESTED** [review]
Very tiny nitpicks.

**Review by cjw296 — APPROVED** [review]

**Pull request merged**